### ISSUE TITLE
Rent charging

### ISSUE THREAD

**Issue by breathx** [gear-tech/gear#646]
### Problem to Solve

Updated representation of #24 .

We have to charge gas/value rent for activities like:
- [x] message keeps in wait-list (already done by @ekovalev)
- [ ] code upload
- [ ] program keeps in storage (I think should be payed on init for some future time. In case of reservation end, disable program changing it's status)
- [ ] Expand wait-list charging after #496 for charging depend of stored data size
- [ ] message keeps in mailbox (partially relates to #642)
_... this list can be updated with new features added_

### Possible Solution

Need to expand offchain worker functionality for all of this cases.

**Comment by SabaunT** [comment]
About `code upload`: I think we should charge for submitting unused code. Metadata for triggering such charge was provided.
Separate issue should be created.

**Comment by breathx** [comment]
Will be applied after #969
